### DISCUSSION TITLE
Generate a simulation for my new Antenna Design

### DISCUSSION THREAD

**Discussion by TahhaR** [thliebig/openEMS-Project#67] [Q&A Usage]
Hello people ,
here is my AppCSXCAD in the bottom. i would like to know how i could perform a good mesh and how could i generate a good script to simulate my antenna which seems to be nearly a patch antenna. i tried to do the same as the posted example but unfortunately it doesn't work as i desired. i joined some pictures.
As well as i tried with differet generated scripts. So please, be welcome to help me out :D .

![Capture333](https://github.com/thliebig/openEMS-Project/assets/61636792/b2c7753b-b090-45ee-9934-94af5a1ac7e0)
![Capture3](https://github.com/thliebig/openEMS-Project/assets/61636792/5d690530-527d-482d-9b93-29d90829f287)

**Comment by LubomirJagos** [comment]
hmmm, maybe this will little confusing for Thorsten as I don't know if he know about GUI addon which I created originally for FreeCAD which is GUI frontent to generate simulation for openEMS in kind a Ansys HFSS style, I wanted to post here about it when I polish it but never mind, it's running quite good now, so just for info this simulation is generated from this addon https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export

It seems almost right, but I see there impedance cube is out of your patch, I guess it's 50Ohm port so your patch has excitation on wrong place and I think that energy is then not floating into your patch, at first I would try to move that port to touch patch and ground.

Then it's needed to make sure that gridlines are crossing input port and also is good to check boundary conditions that they are set to PML or MUR (by default they are set to PEC what is metal around structure, in case of antennas it's wrong).

If this design is not secret could you post here your FreeCAD file and .ini settings file please? I can check it.

for Thorsten: Sorry that this seems to be unrelated but openEMS is so cool and since I discovered it I know that it's first open-source FDTD engine capable to be easy extended and it was missing GUI so I made one and tested it to got it into same level as HFSS to fill the gap in open-source SW since there wasn't no full open-source alternative to HFSS or CST till now
I wanted to thank you for work what you have done because it must be tremendous amount of work to make it at the level like you did (solver core, AppCSXCAD what is editor and viewer for design verification) and to validate that its calculations are correct, create python, octave interfaces and pack that in ecosystem. (there is even much more in openEMS like debeye materials, voxel material, ...)
So to make this project thrive and not to fall apart in future I added GUI frontend to make easy fast way to create complicated simulations (like scripting is cool but it remainds me situation like TeX vs Word, TeX is cool and capable but to work fast and easy Word is more suitable)

Also must mention here MisterHW (https://github.com/MisterHW) who is co-author, collaborator, without his push and commits and reminders this GUI addon will be still in diapers and I wouldn't be motivated to push it into such final stage so fast.

I also opened discussion here for this problems: https://github.com/LubomirJagos/FreeCAD-OpenEMS-Export/discussions

**Comment by VolkerMuehlhaus** [comment]
Have you checked port position, connection and orientation? As an openEMS newbie, I have seen such Energy = 0.00e0 responses when the port was disconnected (wrong place or orientation)

**Reply by thliebig** [reply]
Yeah the energy is all zero, that means you have no (working) excitation...
